### DISCUSSION TITLE
Error [ERR_MODULE_NOT_FOUND] in Nuxt 3 not in Vue 3.

### DISCUSSION THREAD

**Discussion by tereith** [nuxt/nuxt#18115] [Questions] [answered]
Hi!

Testing the same import in Nuxt 3 and Vue 3. In the Vue 3 project it works as expected, but I get this error message in the Nuxt 3 project (and a 500 in the browser). Anyone knows what's going on here?

Nuxt 3:
<img width="1223" alt="Screenshot 2021-11-06 at 12 57 25" src="https://user-images.githubusercontent.com/1519185/140608757-a58b92d9-d041-4c01-b027-ae34a209bccb.png">

Vue 3:
<img width="927" alt="Screenshot 2021-11-06 at 13 07 54" src="https://user-images.githubusercontent.com/1519185/140609039-5fe7ac14-ec49-4630-bca1-f1a8f7278e74.png">

Regards,
Terje Eithun

**Comment by wobsoriano** [comment]
Try to transpile it

```ts
export default defineNuxtConfig({
  build: {
    transpile: ['@schibsted/account-sdk-browser']
  }
})
```

**Reply by tereith** [reply]
Hi! Tried it, but got another 500 and a 'require is not defined in ES module scope, you can use import instead' error.

<img width="644" alt="Screenshot 2021-11-06 at 16 08 58" src="https://user-images.githubusercontent.com/1519185/140614404-ba11de04-910c-4d4f-9efd-79221682229e.png">

**Reply by joshkoppang** [reply]
Anything else we could try here? A big blocker in migrating.

**Comment by joshkoppang** [comment]
Same thing for me with [https://github.com/solana-labs/wallet-adapter](https://github.com/solana-labs/wallet-adapter).

Initially, I get this:
`Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'D:\Source\nuxt3-app1\node_modules\@solana\wallet-adapter-vue\lib\errors' imported from D:\Source\nuxt3-app1\node_modules\@solana\wallet-adapter-vue\lib\index.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:416:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}`

Checking the [index.ts](https://github.com/solana-labs/wallet-adapter/blob/master/packages/core/vue/src/index.ts), I see they are exporting like this:
`export * from './errors';
export * from './useAnchorWallet';
export * from './useLocalStorage';
export * from './useWallet';
export * from './WalletProvider';`

If I follow the chain of transpiling and add each package as I walk through its dependencies, I get the `require is not defined` error.

**Reply by tereith** [reply]
This worked for me: `https://www.npmjs.com/package/@originjs/vite-plugin-commonjs `

**Reply by joshkoppang** [reply]
I tried the following in `vite.config.js` in the root of the project. Doesn't seem to have done anything.

```
import { defineConfig } from 'vite'
import { esbuildCommonjs, viteCommonjs } from '@originjs/vite-plugin-commonjs'

export default defineConfig({
    plugins: [
        viteCommonjs()
    ],
    optimizeDeps:{
        esbuildOptions:{
          plugins:[
            esbuildCommonjs(['@solana/wallet-adapter-vue']) 
          ]
        }
      }
})
```

**Reply by joshkoppang** [reply]
Tried this too in `nuxt.config.ts`, no joy.

```
export default defineNuxtConfig({
    build: {
        transpile: [
        ],
    },
    vite: {
        plugins: [
            viteCommonjs()
        ],
        optimizeDeps:{
            esbuildOptions:{
              plugins:[
                esbuildCommonjs(['@solana/wallet-adapter-vue']) 
              ]
            }
          }
    }
})
```

**Comment by Hugoo** [comment, marked as answer]
I fixed this issue with this: https://v3.nuxtjs.org/concepts/esm#aliasing-libraries

**Reply by Hugoo** [reply]
Well, it actually fixed the compile error but the objects I import from the libs are `undefined`. So maybe not fixed yet 😅

**Reply by tereith** [reply]
You're probably right. :) The import itself doesn't give an error any more, but when trying to instantiate an object with the import it fails:
![Screenshot 2021-12-08 at 07 57 28](https://user-images.githubusercontent.com/1519185/145162933-46de8d94-2885-425c-9a03-9dda077c0d5a.png)
